tests/unit/events/publisher_tests.py

```diff
@@ -0,0 +1,112 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import datetime
+import gzip
+import unittest
+
+import requests
+
+from baseplate import config, metrics
+from baseplate._compat import BytesIO
+from baseplate.events import publisher
+
+from ... import mock
+
+
+class BatcherTests(unittest.TestCase):
+    def setUp(self):
+        self.consumer = mock.Mock()
+        self.consumer.batch_size_overhead = 0
+        self.consumer.get_item_size = lambda item: len(item)
+        self.consumer.batch_size_limit = 5
+        self.batcher = publisher.Batcher(self.consumer)
+
+    def test_flush_empty_does_nothing(self):
+        self.batcher.flush()
+        self.assertEqual(self.consumer.consume_batch.called, False)
+
+    def test_start_time(self):
+        with mock.patch("time.time") as mock_time:
+            self.assertEqual(self.batcher.batch_age, 0)
+
+            mock_time.return_value = 33
+            self.batcher.add("a")
+            self.assertEqual(self.batcher.batch_start, 33)
+
+            mock_time.return_value = 34
+            self.batcher.add("b")
+            self.assertEqual(self.batcher.batch_start, 33)
+
+            mock_time.return_value = 35
+            self.assertEqual(self.batcher.batch_age, 2)
+
+        self.batcher.flush()
+        self.assertEqual(self.batcher.batch_start, None)
+
+    def test_manual_flush(self):
+        self.batcher.add("a")
+        self.batcher.add("b")
+        self.batcher.flush()
+        self.consumer.consume_batch.assert_called_once_with(["a", "b"])
+
+    def test_flush_when_full(self):
```

Comment:
I think what Chad means here is that you specifically specify `7`, even though if someone changed `batch_size_limit` in your test on line 24 to `10`, this would fail. He's wondering if you should avoid that by specifically setting this range to be 1 (or 2) larger than the batch size limit you defined on line 24.